example/build.gradle

```diff
@@ -1,12 +1,106 @@
-apply plugin: 'org.jetbrains.kotlin.platform.native'
+apply plugin: 'kotlin-multiplatform'
+apply plugin: 'java'
 
+kotlin {
+    targets {
+        fromPreset(presets.macosX64, 'macos')
+        fromPreset(presets.linuxX64, 'linux')
+        fromPreset(presets.mingwX64, 'windows')
+        fromPreset(presets.jvm, 'jvm')
+        fromPreset(presets.js, 'js') {
+            compilations.main {
+                compileKotlinJs.kotlinOptions {
+                    moduleKind = "commonjs"
+                }
+            }
+        }
 
-components.main {
-    targets = ['linux', 'macos']
-
-    outputKinds = [EXECUTABLE]
+        configure([macos, linux, windows]) {
+            compilations.main.outputKinds 'executable'
+        }
+    }
 }
 
 dependencies {
-    implementation project(':native')
+    commonMainImplementation 'org.jetbrains.kotlin:kotlin-stdlib'
+    jvmMainImplementation 'org.jetbrains.kotlin:kotlin-stdlib-jdk8'
+    jsMainImplementation 'org.jetbrains.kotlin:kotlin-stdlib-js'
+
+    commonMainImplementation rootProject
+    jvmMainImplementation rootProject
+    jsMainImplementation rootProject
+    macosMainImplementation rootProject
+    linuxMainImplementation rootProject
+    windowsMainImplementation rootProject
+}
+
+['Macos', 'Linux', 'Windows'].forEach { target ->
+    task "run$target" {
+        dependsOn "linkDebugExecutable$target"
+        doLast {
+            exec {
+                commandLine kotlin.targets[target.toLowerCase()].compilations.main.getBinary('executable', 'debug')
+            }
+        }
+    }
+}
+
+def jvmCompilations = kotlin.targets.jvm.compilations
+task createExecutableJar(type: Jar) {
+    dependsOn jvmMainClasses
+    manifest {
+        attributes 'Main-Class': 'ExampleKt'
+    }
+    baseName = 'example'
+    from { jvmCompilations.main.output }
+    from { jvmCompilations.main.runtimeDependencyFiles.collect { it.isDirectory() ? it : zipTree(it) } }
+}
+
+task runJar(type: Exec) {
```

Comment:
I'm having troubles running the examples in Intellij IDEA, because there's no jdk defined in the example project. Couldn't we define all required settings for the java plugin so it's running in the IDE as well?

I tried it with some changes based on the [multiplatform documentation](https://kotlinlang.org/docs/reference/building-mpp-with-gradle.html), but couldn't get it to work. Do you have any idea what to do? Otherwise I'll work on it on another branchPR.

Reply:
Excellent idea, I didn't try.  I'll give it a shot.

Reply:
I think this is possible but I will need to fiddle around with the java plugin some more to get it working correctly.  We should handle this with a different PR.